src/Core/Content/Test/Seo/SeoUrlGeneratorTest.php

```diff
@@ -194,6 +202,90 @@ public function testVariantInheritance(): void
         static::assertEquals('parent/green', $urls[$ids->get('green')]);
     }
 
+    public function testNotBeingStateful(): void
+    {
+        $categoryIds = $this->getCategoryIds(2);
+
+        self::assertCount(2, $categoryIds, 'this is important for the test as you need more items to iterate for a context switch test');
+
+        /** @var \Generator $firstRun */
+        $firstRun = $this->seoUrlGenerator->generate(
+            $categoryIds,
+            'template first run',
+            $this->seoUrlRouteRegistry->findByRouteName(TestNavigationSeoUrlRoute::ROUTE_NAME),
+            $this->salesChannelContext->getContext(),
+            $this->salesChannelContext->getSalesChannel()
+        );
+        /** @var \Generator $secondRun */
+        $secondRun = $this->seoUrlGenerator->generate(
+            $categoryIds,
+            'template second run',
+            $this->seoUrlRouteRegistry->findByRouteName(TestNavigationSeoUrlRoute::ROUTE_NAME),
+            $this->salesChannelContext->getContext(),
+            $this->salesChannelContext->getSalesChannel()
+        );
+
+        /** @var SeoUrlEntity $url */
+        foreach ($firstRun as $url) {
```

Comment:
why foreach, when you could get the first one from $firstRun

Reply:
I need to trigger an iteration of the generator. next would be sufficient as well but eventually same result.